product_assortment/models/ir_filters.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2021 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.osv import expression
+from odoo.tools.safe_eval import datetime, safe_eval
+
+
+class IrFilters(models.Model):
+    _inherit = "ir.filters"
+
+    @api.model
+    def _get_default_is_assortment(self):
+        if self.env.context.get("product_assortment", False):
+            return True
+        return False
+
+    @api.model
+    def _update_assortment_default_values(self, vals_list):
+        """
+        If we create the filter through the assortment, we need to force
+        model_id to product.product
+        """
+        product_assortment = self.env.context.get("product_assortment", False)
+        if not product_assortment:
+            return vals_list
+        model = self.env.ref("product.model_product_product")
+        for vals in vals_list:
+            if not vals.get("model_id"):
+                vals.update({"model_id": model.model})
+        return vals_list
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        self._update_assortment_default_values(vals_list=vals_list)
+        return super().create(vals_list=vals_list)
```

Comment:
It is clearer if we put the field definitions at the beginning of the class and the methods after

Reply:
Done